test/test_distributions.py

```diff
@@ -245,6 +245,19 @@ def _check_sampler_sampler(self, torch_dist, ref_dist, message, multivariate=Fal
             self.assertLess(-threshold, bias, message)
             self.assertLess(bias, threshold, message)
 
+    def _check_sampler_discrete(self, torch_dist, ref_dist, message,
+                                num_samples=10000, failure_rate=1e-3):
+        """Runs a Chi2-test for the support, but ignores tail instead of combining"""
+        torch_samples = torch_dist.sample_n(num_samples).squeeze()
+        if isinstance(torch_samples, Variable):
+            torch_samples = torch_samples.data
+        torch_samples = torch_samples.cpu().numpy()
+        unique, counts = np.unique(torch_samples, return_counts=True)
+        pmf = ref_dist.pmf(unique)
+        msk = (counts > 5) & ((pmf * num_samples) > 5)
```

Comment:
Can you add an assertion like
```py
self.assertGreater(len(msk), 4, "Distribution is too sparse for test; try increasing num_samples")
```
to avoid silent failures?

Reply:
Definitely, that's a great idea.